RummikubGameApp/RummikubGameApp/GameLogic.cs

```diff
@@ -0,0 +1,340 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Drawing;
+using System.Windows.Forms;
+using System.Text.RegularExpressions;
+
+namespace RummikubGameApp
+{
+    public class GameLogic
+    {
+        private Tile[] tiles;
+        public static Tile[] hand;
+        private Tile[] AIhand;
+        private Tile[,] deck;
+        private int currentPlayerIndex;
+        private int tilesIndex;
+
+        public GameLogic()
+        {
+            this.tiles = new Tile[106];
+            hand = new Tile[106];
+            this.AIhand = new Tile[106];
+            this.deck = new Tile[4, 20];
+            this.currentPlayerIndex = 2;
+            this.currentPlayerIndex = 28;
+        }
+
+        public Tile[] StartGame()
+        {
+            // Initialize the game by creating the tiles and shuffling them
+            
+            InitializeTiles();
+            Shuffle();
+            Tile[] arr = Deal();
+            // Set the first player as the current player
+            currentPlayerIndex = 0;
+            return arr;
+        }
+
+
+        public void InitializeTiles()

```

Comment:
- I think InitializeTiles is better written as a method which builds the object and returns it, something like `return tiles` and put it into the `this.tiles = InitializeTiles()` var, what way you won't have leftovers by accident